src/Sema.zig

```diff
@@ -16727,90 +16717,66 @@ fn elemVal(
     sema: *Sema,
     block: *Block,
     src: LazySrcLoc,
-    array: Air.Inst.Ref,
+    indexable: Air.Inst.Ref,
     elem_index_uncasted: Air.Inst.Ref,
     elem_index_src: LazySrcLoc,
 ) CompileError!Air.Inst.Ref {
-    const array_src = src; // TODO better source location
-    const array_ty = sema.typeOf(array);
+    const indexable_src = src; // TODO better source location
+    const indexable_ty = sema.typeOf(indexable);
 
-    if (!array_ty.isIndexable()) {
-        return sema.fail(block, src, "array access of non-indexable type '{}'", .{array_ty});
+    if (!indexable_ty.isIndexable()) {
+        return sema.fail(block, src, "indexing of non-indexable type '{}'", .{indexable_ty});
```

Comment:
I'd prefer to keep calling this `array access` the same way `{ 1, 2 }` is called `array initialization syntax` even though it works for vectors and tuples too, and `.{ .foo = 1 }` is called `struct initialization syntax` when it can also be used for unions.

Reply:
Looking at the docs, we seem to use a mix of "indexing" and "array indexing". It also refers to "array initialization syntax" instead as "array literals" and "list literals". I don't mind keeping it as "array access", but I also would like to stay consistent where possible.